### DISCUSSION TITLE
Dusk setstatus

### DISCUSSION THREAD

**Discussion by kmarzic** [bakkeby/dusk#17] [General]
Hi,

From shell script, I am running the following:
```
$ while [ 1 ]; do duskc run_command setstatus 0 "$(date +%T)"; sleep 1; done
```
The time on the status bar only refreshes when I change workspaces, not every second. 
Do I need to somehow specify this behavior in `config.h` file in `static const BarRule barrules[]`?

Thanks!

**Comment by bakkeby** [comment]
I tried with a straight clone, make, then starting dusk through Xephyr and running your script. It updates every second for me so I can't quite replicate this behaviour.

Does killing the compositor make any difference? Just curious.

**Reply by kmarzic** [reply]
Hi,

Many thanks for trying to resolve my configuration problem!

I have the following setup:
- using `ly` display manager and running `dusk` or `dwm` from `~/.xsession`:
```
$ cat ~/.xsesssion
...
~/bin/status.sh &
...
while true;
do
    ## dusk
    ${HOME}/bin/dusk 2> ~/.dusk.log
    ## dwm
    # ${HOME}/bin/dwm 2> ~/.dwm.log
done
...
```

```
$ cat ~/bin/status.sh
...
while true
do
    ## dusk
    duskc run_command setstatus 0 "$(date +%T)"
    ## dwm
    # xsetroot -name "$(date +%T)"
    sleep 1
done
...
```

- using `picom`, running from `~/.xprofile`

Effects:
1. running `dwm` wih `picom`
    - time in status bar **is** refreshed every second
2. running `dusk` with `picom`
    - time in status bar **is not** refreshed every second (is refreshed only when i switch to another workspace) 
3. running `dusk` with `picom`, and then killing `picom` process
    - time in status bar **is not** refreshed every second (is refreshed only when i switch to another workspace) 

Best regards

**Reply by bakkeby** [reply]
Can you confirm if alternating the status makes it update continuously?

```sh
$ cat ~/bin/status.sh
...
while true
do
    duskc run_command setstatus 0 "$(date +%T)"
    sleep 0.5
    duskc run_command setstatus 0 "hello"
    sleep 0.5
done
...
```

**Reply by kmarzic** [reply]
Hi,

After putting `sleep 0.5` after each `dusck` in `while loop`, situation is the same:
status bar is refreshed only when I switch to another workspace.

**Reply by bakkeby** [reply]
The `sleep 0.5` was just to have two updates within one second. The main purpose of that test was to see if the status changing size makes any difference, hence the "hello".

What distribution are you on?

The way you are running dwm / dusk via .xsession is pretty weird, doesn't that override whatever you choose via the display manager?

**Comment by kmarzic** [comment]
Hi,
I am using Arch Linux (rolling Linux distribution, last update few days ago).
Similar setup I have on Ubuntu 22.04 LTS (only difference is `gdm3` instead of `ly`).

**Reply by bakkeby** [reply]
I tried this with arch and `ly` and a similar setup in `.xsession` and I can't reproduce the issue. Curious as to what is the cause of it on your end.

**Comment by kmarzic** [comment]
Hi,
I managed to solve problem.
I added `duskc run_command xrdb` to `~/bin/status.sh` (I found it somewhere on your wiki pages).

```
$ cat ~/bin/status.sh
...
while true
do
    duskc run_command setstatus 0 "$(date +%T)"
    duskc run_command xrdb
    sleep 1
done
````

Thanks for your help!
